### PR TITLE
S3655: Prepare TestCases

### PR DESCRIPTION
**Pull request by antonioaversa**
Task 1 of #6794

### CODE REVIEW

**Review by zsolt-kolbay-sonarsource — APPROVED** [review]
LGTM
There are a couple of test cases which I don't fully understand, but they weren't introduced in this PR.

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/EmptyNullableValueAccess.cs [hunk 1]
```diff
@@ -0,0 +1,315 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+public class EmptyNullableValueAccess
+{
+    protected void LogFailure(Exception e)
```
[the remark this answers is not in the document]
Comment (pavel-mikula-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/EmptyNullableValueAccess.cs:7]: Based on commit history, this was supposed to prove that CFG is build for `try` block.

The test itself could be removed. It has nothing to do with nullable values.

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/EmptyNullableValueAccess.cs [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+        }
+
+        if (o == nameof(Object))
+        {
+            o.ToString(); // Compliant
+        }
+    }
+
+    private int i0;
+    public void SetI0()
```
[the remark this answers is not in the document]
Comment (pavel-mikula-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/EmptyNullableValueAccess.cs:42]: Probably some debugging leftovers from the time the old CFG or SE was build.

Can be safely removed

**Review by pavel-mikula-sonarsource — APPROVED** [review]
LGTM

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/EmptyNullableValueAccess.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (antonioaversa) [reply analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/EmptyNullableValueAccess.cs:7]: Removed

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/EmptyNullableValueAccess.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (antonioaversa) [reply analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/EmptyNullableValueAccess.cs:42]: Both the `SetI0` method and its `i0` have been removed.

**Pull request merged**